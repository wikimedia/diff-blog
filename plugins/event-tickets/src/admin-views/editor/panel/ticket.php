<?php
$is_admin = tribe_is_truthy( tribe_get_request_var( 'is_admin', is_admin() ) );

if ( ! isset( $post_id ) ) {
	$post_id = get_the_ID();
}

$provider       = null;
$provider_class = null;
$ticket         = null;

if ( ! isset( $ticket_id ) ) {
	$ticket_id = null;

	if ( ! $is_admin ) {
		$provider_class = Tribe__Tickets__Tickets::get_default_module();
	}
} else {
	$provider = tribe_tickets_get_ticket_provider( $ticket_id );

	if ( ! empty( $provider ) ) {
		$provider_class = $provider->class_name;

		$ticket = $provider->get_ticket( $post_id, $ticket_id );
	}
}

$modules = Tribe__Tickets__Tickets::modules();

$default_module_class = empty( $default_module ) ? '' : $default_module->class_name;
?>

<div id="tribe_panel_edit" class="ticket_panel panel_edit tribe-validation" aria-hidden="true" data-default-provider="<?php echo esc_attr( $default_module_class ); ?>">
	<?php
	/**
	 * Allows for the insertion of additional elements into the main ticket edit panel.
	 *
	 * @since 4.6
	 *
	 * @param int Post ID
	 * @param int Ticket ID
	 */
	do_action( 'tribe_events_tickets_pre_edit', $post_id, $ticket_id );
	?>

	<div id="ticket_form" class="ticket_form tribe_sectionheader tribe-validation">
		<div id="ticket_form_table" class="eventtable ticket_form">
			<div
				class="tribe-dependent"
				data-depends="#Tribe__Tickets__RSVP_radio"
				data-condition-is-not-checked
			>
				<h4
					id="ticket_title_add"
					class="ticket_form_title tribe-dependent"
					data-depends="#ticket_id"
					data-condition-is-empty
				>
					<?php
					echo esc_html(
						sprintf(
							_x( 'Add new %s', 'admin add new ticket panel heading', 'event-tickets' ), tribe_get_ticket_label_singular_lowercase( 'admin_add_new_ticket_panel_heading' )
						)
					); ?>
				</h4>
				<h4
					id="ticket_title_edit"
					class="ticket_form_title tribe-dependent"
					data-depends="#ticket_id"
					data-condition-is-not-empty
				>
					<?php
					echo esc_html(
						sprintf(
							_x( 'Edit %s', 'admin edit ticket panel heading', 'event-tickets' ),
							tribe_get_ticket_label_singular_lowercase( 'admin_edit_ticket_panel_heading' )
						)
					); ?>
				</h4>
			</div>
			<div
				class="tribe-dependent"
				data-depends="#Tribe__Tickets__RSVP_radio"
				data-condition-is-checked
			>
				<h4
					id="rsvp_title_add"
					class="ticket_form_title tribe-dependent"
					data-depends="#ticket_id"
					data-condition-is-empty
				>
					<?php
					echo esc_html(
						sprintf(
							_x( 'Add new %s', 'admin add new ticket panel heading', 'event-tickets' ),
							tribe_get_rsvp_label_singular( 'admin_add_new_ticket_panel_heading' )
						)
					); ?>
				</h4>
				<h4
					id="rsvp_title_edit"
					class="ticket_form_title tribe-dependent"
					data-depends="#ticket_id"
					data-condition-is-not-empty
				>
					<?php
					echo esc_html(
						sprintf(
							_x( 'Edit %s', 'admin edit ticket panel heading', 'event-tickets' ),
							tribe_get_rsvp_label_singular( 'admin_edit_ticket_panel_heading' )
						)
					); ?>
				</h4>
			</div>
			<section id="ticket_form_main" class="main">
				<div class="input_block">
					<label class="ticket_form_label ticket_form_left" for="ticket_name"><?php esc_html_e( 'Type:', 'event-tickets' ); ?></label>
					<input
						type='text'
						id='ticket_name'
						name='ticket_name'
						class="ticket_field ticket_form_right"
						size='25'
						value="<?php echo esc_attr( $ticket ? $ticket->name : null ); ?>"
						data-validation-is-required
						data-validation-error="<?php echo esc_attr( sprintf( _x( '%s type is a required field', 'admin edit ticket panel error', 'event-tickets' ), tribe_get_rsvp_label_singular( 'admin_edit_ticket_panel_error' ) ) ); ?>"
					/>
					<span
						class="tribe_soft_note ticket_form_right"
						data-depends="#Tribe__Tickets__RSVP_radio"
						data-condition-not-checked
					><?php
						echo esc_html(
							sprintf(
								_x( '%1$s type name shows on the front end and emailed %2$s', 'admin edit ticket panel note', 'event-tickets' ),
								tribe_get_ticket_label_singular( 'admin_edit_ticket_panel_note' ),
								tribe_get_ticket_label_plural_lowercase( 'admin_edit_ticket_panel_note' )
							)
						); ?>
					</span>
					<span
						class="tribe_soft_note ticket_form_right"
						data-depends="#Tribe__Tickets__RSVP_radio"
						data-condition-is-checked
					><?php
						echo esc_html(
							sprintf(
								_x( '%1$s type name shows on the front end and emailed %2$s', 'admin edit ticket panel note', 'event-tickets' ),
								tribe_get_rsvp_label_singular( 'admin_edit_rsvp_panel_note' ),
								tribe_get_rsvp_label_plural_lowercase( 'admin_edit_ticket_panel_note' )
							)
						); ?>
					</span>
				</div>
				<fieldset id="tribe_ticket_provider_wrapper" class="input_block" aria-hidden="true" >
					<legend class="ticket_form_label"><?php esc_html_e( 'Sell using:', 'event-tickets' ); ?></legend>
					<?php foreach ( $modules as $class => $module ) : ?>
						<input
							type="radio"
							name="ticket_provider"
							id="<?php echo esc_attr( $class . '_radio' ); ?>"
							value="<?php echo esc_attr( $class ); ?>"
							class="ticket_field ticket_provider"
							tabindex="-1"
							<?php checked( true, $provider_class === $class ); ?>
						>
						<span>
							<?php
							/**
							 * Allows for the editing of the module name before output
							 *
							 * @since 4.6
							 *
							 * @param string $module the module name
							 */
							echo esc_html( apply_filters( 'tribe_events_tickets_module_name', $module ) );
							?>
						</span>
					<?php endforeach; ?>
				</fieldset>
				<?php
				/**
				 * Allows for the insertion of additional content into the ticket edit form - main section
				 *
				 * @since 4.6
				 *
				 * @param int Post ID
				 * @param int Ticket ID
				 */
				do_action( 'tribe_events_tickets_metabox_edit_main', $post_id, $ticket_id ); ?>
			</section>

			<div class="accordion">
				<?php
				/** @var Tribe__Tickets__Admin__Views $admin_views */
				$admin_views = tribe( 'tickets.admin.views' );

				$admin_context = [
					'post_id'   => $post_id,
					'ticket_id' => $ticket_id,
					'provider'  => $provider,
				];

				$admin_views->template( 'editor/fieldset/advanced', $admin_context );
				$admin_views->template( 'editor/fieldset/history', [ 'post_id' => $post_id, 'ticket_id' => $ticket_id ] );

				/**
				 * Allows for the insertion of additional content sections into the ticket edit form accordion
				 *
				 * @since 4.6
				 *
				 * @param int Post ID
				 * @param int Ticket ID
				 */
				do_action( 'tribe_events_tickets_metabox_edit_accordion_content', $post_id, $ticket_id );
				?>
			</div>

			<?php
			/**
			 * Allows for the insertion of additional elements into the main ticket edit panel below the accordion section
			 *
			 * @since 4.6
			 *
			 * @param int Post ID
			 * @param int Ticket ID
			 */
			do_action( 'tribe_events_tickets_post_accordion', $post_id, $ticket_id );
			?>
			<div class="ticket_bottom">
				<input
					type="hidden"
					name="ticket_id"
					id="ticket_id"
					class="ticket_field"
					value="<?php echo esc_attr( $ticket_id ); ?>"
				/>
				<input
					type="button"
					id="ticket_form_save"
					class="button-primary tribe-dependent tribe-validation-submit"
					name="ticket_form_save"
					value="<?php echo esc_attr( sprintf( _x( 'Save %s', 'meta box ticket form button text', 'event-tickets' ), tribe_get_ticket_label_singular_lowercase( 'meta_box_ticket_form_button_text' ) ) ); ?>"
					data-depends="#Tribe__Tickets__RSVP_radio"
					data-condition-is-not-checked
				/>
				<input
					type="button"
					id="rsvp_form_save"
					class="button-primary tribe-dependent tribe-validation-submit"
					name="ticket_form_save"
					value="<?php echo esc_attr( sprintf( _x( 'Save %s', 'RSVP form save value', 'event-tickets' ), tribe_get_rsvp_label_singular( 'form_save_value' ) ) ); ?>"
					data-depends="#Tribe__Tickets__RSVP_radio"
					data-condition-is-checked
				/>
				<input
					type="button"
					id="ticket_form_cancel"
					class="button-secondary"
					name="ticket_form_cancel"
					value="<?php esc_attr_e( 'Cancel', 'event-tickets' ); ?>"
				/>

				<?php
				/**
				 * Allows for the insertion of additional content into the ticket edit form bottom (buttons) section
				 *
				 * @since 4.6
				 *
				 * @param int Post ID
				 * @param int Ticket ID
				 */
				do_action( 'tribe_events_tickets_bottom', $post_id, $ticket_id );
				?>

				<div id="ticket_bottom_right">
					<?php
					/**
					 * Allows for the insertion of additional content into the ticket edit form bottom (links on right) section
					 *
					 * @since 4.6
					 *
					 * @param int Post ID
					 * @param int Ticket ID
					 */
					do_action( 'tribe_events_tickets_bottom_right', $post_id, $ticket_id );
					?>
				</div>
			</div>
		</div>
	</div>
</div>
