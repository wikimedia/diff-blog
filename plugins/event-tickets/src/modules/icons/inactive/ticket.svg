<svg width="60" height="60" xmlns="http://www.w3.org/2000/svg"><g fill="#AEB4BB" fill-rule="evenodd"><path d="M14.267 31.31l-.434-1.434a.25.25 0 0 1 .167-.311l28.5-8.614a.251.251 0 0 1 .312.167l.434 1.436a.25.25 0 0 1-.167.312l-28.5 8.613a.251.251 0 0 1-.312-.168M16.66 39.05l-.435-1.435a.25.25 0 0 1 .167-.311l28.5-8.613a.25.25 0 0 1 .312.167l.434 1.435a.25.25 0 0 1-.167.312l-28.5 8.613a.251.251 0 0 1-.312-.168"/><path d="M12.812 46.805l-.061-.23-.237-.91-.66-2.528a.416.416 0 0 1-.01-.061.305.305 0 0 1 .039-.186c.074-.112.115-.174.153-.237a4.164 4.164 0 0 0 .586-2.292 4.13 4.13 0 0 0-.27-1.297c-.367-.956-1.066-1.7-1.971-2.095a.307.307 0 0 1-.181-.21l-.32-1.223a.347.347 0 0 1 .054-.287 4.097 4.097 0 0 0 .438-3.777c-.37-.958-1.07-1.702-1.971-2.095a.314.314 0 0 1-.18-.206l-.877-3.372c-.025-.098-.085-.292-.085-.292l.294-.1 39.046-11.925.28-.08.082.3.81 3.112a.336.336 0 0 1-.065.298 4.01 4.01 0 0 0-.861 1.622 4.065 4.065 0 0 0 .329 2.939c.476.896 1.25 1.537 2.186 1.808a.297.297 0 0 1 .2.217l.186.706a.32.32 0 0 1-.066.292c-.867.99-1.197 2.41-.862 3.704.339 1.292 1.304 2.313 2.517 2.666a.301.301 0 0 1 .203.217l.806 3.107.08.306s-.195.062-.29.094l-39.03 11.929-.292.086zM54.578 35.1l-1.364-5.24c-.115-.446-.5-.753-.931-.755-.725-.001-1.376-.537-1.57-1.287-.196-.747.098-1.562.717-1.982a1.1 1.1 0 0 0 .432-1.173l-.624-2.397c-.116-.444-.483-.753-.931-.754a1.645 1.645 0 0 1-1.372-.814 1.86 1.86 0 0 1-.216-1.338 1.79 1.79 0 0 1 .735-1.117 1.09 1.09 0 0 0 .43-1.173l-1.364-5.245-.035-.12-.037-.112c-.024-.065-.033-.098-.048-.134-.037-.086-.042-.098-.051-.114a.763.763 0 0 0-.098-.16.572.572 0 0 0-.126-.112.675.675 0 0 0-.171-.065c-.13-.008-.157-.01-.193-.008-.12.014-.145.019-.175.024l-.17.037c-.015.003-.048.012-41.605 12.709-.854.296-.932.446-.694 1.338l1.399 5.358c.102.389.408.68.779.742.378.069.723.274.974.577.296.359.439.82.408 1.296a1.845 1.845 0 0 1-.571 1.233 1.102 1.102 0 0 0-.31 1.075l.697 2.67c.1.387.407.68.779.743.376.065.722.273.974.577a1.823 1.823 0 0 1 .408 1.295 1.852 1.852 0 0 1-.107.517 1.897 1.897 0 0 1-.143.313c-.09.15-.198.286-.323.404a1.042 1.042 0 0 0-.299.489 1.13 1.13 0 0 0-.008.585l1.397 5.373c.104.345.173.562.287.693a.407.407 0 0 0 .295.149.801.801 0 0 0 .11 0c.05-.002.109-.01.174-.023.06-.011.128-.026.215-.048l.34-.103 4.59-1.4c.202-.06.407-.124.618-.186l12.233-3.732 1.79-.547 4.057-1.237c.3-.092.598-.183.897-.272l17.106-5.22c.826-.29.867-.48.695-1.33z"/></g></svg>