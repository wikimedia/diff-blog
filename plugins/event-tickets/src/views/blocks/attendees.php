<?php
/**
 * Block: Attendees List
 *
 * Override this template in your own theme by creating a file at:
 * [your-theme]/tribe/tickets/blocks/attendees.php
 *
 * See more documentation about our Blocks Editor templating system.
 *
 * @link https://m.tri.be/1amp Help article for RSVP & Ticket template files.
 *
 * @since 4.9
 * @since 4.11.3 Fix malformed opening `<div>` tag.
 *
 * @version 4.11.3
 */

$title     = $this->attr( 'title' );
$attendees = $this->get( 'attendees', null );
$classes   = [ 'tribe-block', 'tribe-block__attendees' ];

if ( ! is_array( $attendees ) ) {
	return;
}
?>
<div id="tribe-block__attendees" <?php tribe_classes( $classes ); ?>>

	<?php $this->template( 'blocks/attendees/title', [ 'title' => $title ] ); ?>

	<?php $this->template( 'blocks/attendees/description', [ 'attendees' => $attendees ] ); ?>

	<?php foreach ( $attendees as $key => $attendee ) : ?>

		<?php $this->template( 'blocks/attendees/gravatar', [ 'attendee' => $attendee ] ); ?>

	<?php endforeach; ?>
</div>
