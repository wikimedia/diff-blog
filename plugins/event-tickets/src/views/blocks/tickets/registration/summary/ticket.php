<?php
/**
 * Block: Tickets
 * Registration Summary Ticket
 *
 * Override this template in your own theme by creating a file at:
 * [your-theme]/tribe/tickets/blocks/tickets/registration/summary/ticket.php
 *
 * See more documentation about our Blocks Editor templating system.
 *
 * @link https://m.tri.be/1amp Help article for RSVP & Ticket template files.
 *
 * @since 4.9
 * @version 4.11.0
 *
 */
?>
<div class="tribe-tickets__registration__tickets__item">

	<?php $this->template( 'blocks/tickets/registration/summary/ticket-icon', array( 'ticket' => $ticket, 'key' => $key ) ); ?>

	<?php $this->template( 'blocks/tickets/registration/summary/ticket-quantity', array( 'ticket' => $ticket, 'key' => $key ) ); ?>

	<?php $this->template( 'blocks/tickets/registration/summary/ticket-title', array( 'ticket' => $ticket, 'key' => $key ) ); ?>

	<?php $this->template( 'blocks/tickets/registration/summary/ticket-price', array( 'ticket' => $ticket, 'key' => $key ) ); ?>

</div>
