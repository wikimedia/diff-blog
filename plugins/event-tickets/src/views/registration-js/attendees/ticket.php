<?php
/**
 * This template renders the attendee registration block for each ticket
 *
 * Override this template in your own theme by creating a file at:
 * [your-theme]/tribe/tickets/registration-js/attendees/ticket.php
 *
 * @since 4.11.0
 * @since 4.11.3 Require Event Tickets Plus for the template.
 *
 * @version 4.11.3
 *
 */
 if (
	! class_exists( 'Tribe__Tickets_Plus__Meta' )
	|| ! class_exists( 'Tribe__Tickets_Plus__Meta__Storage' )
) {
	return;
}
/**
* @var Tribe__Tickets_Plus__Meta $meta
*/
$meta    = tribe( 'tickets-plus.main' )->meta();
?>
<script type="text/html" id="tmpl-tribe-registration--<?php echo esc_attr( $ticket['id'] ); ?>">
	<?php
	$ticket_qty = $ticket['qty'];
	$post       = get_post( $ticket['id'] );
	$fields     = $meta->get_meta_fields_by_ticket( $post->ID );
	$saved_meta = $storage->get_meta_data_for( $post->ID );
	?>
	<h3 class="tribe-common-h5 tribe-common-h5--min-medium tribe-common-h--alt tribe-ticket__heading"><?php echo esc_html( get_the_title( $post->ID ) ); ?></h3>
	<?php // go through each attendee ?>
	<?php while ( 0 < $ticket_qty ) : ?>
		<?php
			$args = array(
				'event_id'   => $event_id,
				'ticket'     => $post,
				'fields'     => $fields,
				'saved_meta' => $saved_meta,
			);

			$this->template( 'registration-js/attendees/fields', $args );
			$ticket_qty--;
		?>
	<?php endwhile; ?>
</script>
