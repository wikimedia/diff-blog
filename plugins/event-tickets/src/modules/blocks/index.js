/**
 * Wordpress dependencies
 */
import { registerBlockType } from '@wordpress/blocks';

/**
 * Internal dependencies
 */
import { initStore } from '@moderntribe/tickets/data';
import rsvp from '@moderntribe/tickets/blocks/rsvp';
import tickets from '@moderntribe/tickets/blocks/tickets';
import ticket from '@moderntribe/tickets/blocks/ticket';
import attendees from '@moderntribe/tickets/blocks/attendees';

const blocks = [
	rsvp,
	tickets,
	ticket,
	attendees,
];

blocks.forEach( ( block ) => registerBlockType( `tribe/${ block.id }`, block ) );

// Initialize AFTER blocks are registered
// to avoid plugin shown as available in reducer
// but not having block available for use
initStore();

export default blocks;
