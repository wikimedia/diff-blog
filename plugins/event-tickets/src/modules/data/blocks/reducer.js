/**
 * External dependencies
 */
import { combineReducers } from 'redux';

/**
 * Internal dependencies
 */
import rsvp from './rsvp';
import ticket from './ticket';
import attendees from './attendees';

export default combineReducers( {
	rsvp,
	ticket,
	attendees,
} );
