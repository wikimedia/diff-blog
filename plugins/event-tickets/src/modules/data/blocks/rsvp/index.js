/**
 * Internal dependencies
 */
import * as types from './types';
import * as actions from './actions';
import * as selectors from './selectors';
import * as thunks from './thunks';
import reducer from './reducer';
import sagas from './sagas';

export default reducer;
export { types, actions, sagas, selectors, thunks };
