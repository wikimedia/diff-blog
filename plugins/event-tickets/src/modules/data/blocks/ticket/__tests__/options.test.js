/**
 * Internal dependencies
 */
import { options } from '@moderntribe/tickets/data/blocks/ticket';

describe( 'Ticket Options', () => {
	const keys = Object.keys( options );

	keys.forEach( ( key ) => {
		test( key, () => {
			expect( options[ key ] ).toMatchSnapshot();
		} );
	} );
} );
