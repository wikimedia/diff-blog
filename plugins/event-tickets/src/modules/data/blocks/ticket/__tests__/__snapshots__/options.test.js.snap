// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Ticket Options CAPACITY_TYPE_OPTIONS 1`] = `
Array [
  Object {
    "label": "Share capacity with other tickets",
    "value": "capped",
  },
  Object {
    "label": "Set capacity for this ticket only",
    "value": "own",
  },
  Object {
    "label": "Unlimited",
    "value": "unlimited",
  },
]
`;
